test/integration/scheduler/util.go

```diff
@@ -174,71 +175,79 @@ func initTestSchedulerWithOptions(
+	configFactory := factory.NewConfigFactory(testContext.schedulerConfigArgs)
 
 	var err error
 
 	if policy != nil {
-		context.schedulerConfig, err = configFactory.CreateFromConfig(*policy)
+		testContext.schedulerConfig, err = configFactory.CreateFromConfig(*policy)
 	} else {
-		context.schedulerConfig, err = configFactory.Create()
+		testContext.schedulerConfig, err = configFactory.Create()
 	}
 
 	if err != nil {
 		t.Fatalf("Couldn't create scheduler config: %v", err)
 	}
 
 	// set DisablePreemption option
-	context.schedulerConfig.DisablePreemption = disablePreemption
+	testContext.schedulerConfig.DisablePreemption = disablePreemption
 	eventBroadcaster := events.NewBroadcaster(&events.EventSinkImpl{
-		Interface: context.clientSet.EventsV1beta1().Events(""),
+		Interface: testContext.clientSet.EventsV1beta1().Events(""),
 	})
-	context.schedulerConfig.Recorder = eventBroadcaster.NewRecorder(
+	testContext.schedulerConfig.Recorder = eventBroadcaster.NewRecorder(
 		legacyscheme.Scheme,
 		v1.DefaultSchedulerName,
 	)
 
-	context.scheduler = scheduler.NewFromConfig(context.schedulerConfig)
+	testContext.scheduler = scheduler.NewFromConfig(testContext.schedulerConfig)
 
-	scheduler.AddAllEventHandlers(context.scheduler,
+	scheduler.AddAllEventHandlers(testContext.scheduler,
 		v1.DefaultSchedulerName,
-		context.informerFactory.Core().V1().Nodes(),
+		testContext.informerFactory.Core().V1().Nodes(),
 		podInformer,
-		context.informerFactory.Core().V1().PersistentVolumes(),
-		context.informerFactory.Core().V1().PersistentVolumeClaims(),
-		context.informerFactory.Core().V1().Services(),
-		context.informerFactory.Storage().V1().StorageClasses(),
-		context.informerFactory.Storage().V1beta1().CSINodes(),
+		testContext.informerFactory.Core().V1().PersistentVolumes(),
+		testContext.informerFactory.Core().V1().PersistentVolumeClaims(),
+		testContext.informerFactory.Core().V1().Services(),
+		testContext.informerFactory.Storage().V1().StorageClasses(),
+		testContext.informerFactory.Storage().V1beta1().CSINodes(),
 	)
 
 	// set setPodInformer if provided.
 	if setPodInformer {
-		go podInformer.Informer().Run(context.schedulerConfig.StopEverything)
-		cache.WaitForNamedCacheSync("scheduler", context.schedulerConfig.StopEverything, podInformer.Informer().HasSynced)
+		go podInformer.Informer().Run(testContext.schedulerConfig.StopEverything)
+		cache.WaitForNamedCacheSync("scheduler", testContext.schedulerConfig.StopEverything, podInformer.Informer().HasSynced)
 	}
 
 	stopCh := make(chan struct{})
 	eventBroadcaster.StartRecordingToSink(stopCh)
 
-	context.informerFactory.Start(context.schedulerConfig.StopEverything)
-	context.informerFactory.WaitForCacheSync(context.schedulerConfig.StopEverything)
+	testContext.informerFactory.Start(testContext.schedulerConfig.StopEverything)
+	testContext.informerFactory.WaitForCacheSync(testContext.schedulerConfig.StopEverything)
+
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
We should tweak `type testContext struct` to embed a `ctx context.Context` instead of `stopCh chan struct{}` and then reuse that `ctx` here.

Reply:
That would be the anti-pattern. But I agree that we should get rid of the stopCh.

Also, we should consider changing the name of `testContext` to avoid confusion. I don't have many ideas: `testResources` perhaps?